pkg/apiserver/rest/dualwriter_mode2_test.go

```diff
@@ -79,7 +90,7 @@ func TestMode2_Create(t *testing.T) {
 			assert.Equal(t, exampleObj, obj)
 			accessor, err := meta.Accessor(obj)
 			assert.NoError(t, err)
-			assert.Equal(t, accessor.GetResourceVersion(), "")
+			assert.Equal(t, "1", accessor.GetResourceVersion())
```

Comment:
AFAIK `Create` should always return a resource version. Mode1 is already testing that we do. Could it be that this test was not ok? If so, we caught one here.

Reply:
hm, I think in mode2 it should not return a resource_version. Mode1 does return one, because it should always return the legacy object. But in mode2, we return the entity from unified storage, and to follow whatever k8s does, we remove the resource_version for newly created objects. You can see what should be the expected behaviour for each scenario on [these tests](https://github.com/grafana/grafana/blob/init_dualwriter_metrics/pkg/apiserver/rest/dualwriter_mode2_test.go#L505).